code/modules/shuttle/supply.dm

```diff
@@ -223,8 +226,44 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 		msg += export_text + "\n"
 		D.adjust_money(ex.total_value[E])
 
+	SSeconomy.export_total += (D.account_balance - presale_points)
 	SSshuttle.centcom_message = msg
 	investigate_log("Shuttle contents sold for [D.account_balance - presale_points] credits. Contents: [ex.exported_atoms ? ex.exported_atoms.Join(",") + "." : "none."] Message: [SSshuttle.centcom_message || "none."]", INVESTIGATE_CARGO)
 
+/// Generates a box of mail depending on our exports and imports.
+/obj/docking_port/mobile/supply/proc/mail()
+	if(SSeconomy.mail_waiting <= 0)
+		return FALSE
+
+	var/mail_recipients = list()
+	for(var/mob/living/carbon/human/H in shuffle(GLOB.alive_mob_list))
+		if(!H.client || H.stat == DEAD)
+			continue
+		mail_recipients += list(H)
+
+	if(LAZYLEN(mail_recipients) <= 0)
+		return FALSE
+
+	var/list/mail = list()
+	var/obj/structure/closet/crate/mail/mailcrate = new /obj/structure/closet/crate/mail
+	for(var/MI = 0, MI < SSeconomy.mail_waiting, MI++)
+		var/obj/item/mail/NM = new /obj/item/mail(mailcrate)
+		NM.initialize_for_recipient(pick(mail_recipients))
+		mail += list(NM)
+
+	mailcrate.update_icon()
+	SSeconomy.mail_waiting = 0
+
+	var/list/empty_turfs = list()
+	for(var/place in shuttle_areas)
+		var/area/shuttle/shuttle_area = place
+		for(var/turf/open/floor/T in shuttle_area)
+			if(T.is_blocked_turf())
+				continue
+			empty_turfs += T
+
+	mailcrate.forceMove(pick_n_take(empty_turfs))
```

Comment:
why is this a pick n take
Why doesnt this just pass the spawn location directly to the crate

Reply:
It's copied from the above cargo crate generation. This is to pick a random free turf for it to be placed on.